### PR TITLE
chore: add debug logs around shutdownCh

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Add additional debug log entries around `shutdownCh` to allow for verification that SIGINT/SIGTERM signals are being properly sent.

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
@enocom feel free to leave suggestions for additional spots I could add debug log entries and I'll happily add them 😄

**Comment by enocom** [comment]
There are a few other spots. Just adding a logging message directly above these lines will be good enough I think.

https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1026

https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1130

https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1156

https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1177

**Comment by jackwotherspoon** [comment]
> There are a few other spots. Just adding a logging message directly above these lines will be good enough I think.
> 
> https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1026
> 
> https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1130
> 
> https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1156
> 
> https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/b4076f38c462b9a60e8885bbed2116e4f6363517/cmd/root.go#L1177

The last two don't have access to the logger so I couldn't add them...

File: cmd/root.go [hunk 1]
```diff
@@ -1023,6 +1025,7 @@ func runSignalWrapper(cmd *Command) (err error) {
 		defer close(startCh)
 		p, err := proxy.NewClient(ctx, cmd.dialer, cmd.logger, cmd.conf)
 		if err != nil {
+			cmd.logger.Debugf("Error starting proxy. Writing error to shutdown channel.")
```
Comment (enocom) [inline cmd/root.go:1028]: nit: cmd.logger.Debugf(Error starting proxy: %v", err)

File: cmd/root.go [hunk 2]
```diff
@@ -1131,7 +1134,10 @@ func runSignalWrapper(cmd *Command) (err error) {
 		)
 	}
 
-	go func() { shutdownCh <- p.Serve(ctx, notifyStarted) }()
+	go func() {
+		cmd.logger.Debugf("Starting to listen on shutdown channel.")
```
Comment (enocom) [inline cmd/root.go:1138]: By logging the error, we can see what Serve might be complaining about and then infer if the error is reported through the channel based on subsequent log messages.

``` golang
err := <- p.Serve(ctx, notifyStarted)
cmd.logger.Debugf("proxy server error: %v", err)
shutdownCh <- err
```

Comment (jackwotherspoon) [reply cmd/root.go]: did `err := p.Serve(ctx, notifyStarted)` instead of `err := <- p.Serve(ctx, notifyStarted)`

Comment (enocom) [reply cmd/root.go]: Yep you're correct.

**Pull request merged**